drivers/i2c/i2c_ll_stm32_v2.c

```diff
@@ -207,17 +220,26 @@ int i2c_stm32_slave_register(const struct device *dev,
 		return ret;
 	}
 
-	data->slave_cfg = config;
-
 	LL_I2C_Enable(i2c);
 
-	LL_I2C_SetOwnAddress1(i2c, config->address << 1,
-			      LL_I2C_OWNADDRESS1_7BIT);
-	LL_I2C_EnableOwnAddress1(i2c);
+	if (!data->slave_cfg) {
+		data->slave_cfg = config;
 
-	data->slave_attached = true;
+		LL_I2C_SetOwnAddress1(i2c, config->address << 1,
```

Comment:
```suggestion
		LL_I2C_SetOwnAddress1(i2c, config->address << 1U,
```

Reply:
This comes from the original code, but your point is valid, I'll fix it.